### PR TITLE
Added flowcell purpose to mapper

### PR DESCRIPTION
**Pull request by jbeast**
Added new message for flowcell

### CODE REVIEW

File: src/perl/lib/wtsi_clarity/mq/messages/flowcell/control.pm [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+=head1 SYNOPSIS
+
+  my $control = wtsi_clarity::mq::messages::flowcell::control->new(
+    sample_uuid                => '00000000-0000-0000-00000003',
+    study_uuid                 => '00000000-0000-0000-00000004',
+    tag_index                  => 3,
+    entity_type                => 'library_indexed_spike',
+    tag_sequence               => 'ATAG',
+    tag_set_id_lims            => '2',
```
Comment (ke4) [inline src/perl/lib/wtsi_clarity/mq/messages/flowcell/control.pm]: I'm not sure where this attribute is being defined? I could not find it in the code.

File: src/perl/lib/wtsi_clarity/mq/messages/flowcell/control.pm [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+=head1 DEPENDENCIES
+
+=over
+
+=item Moose
+
+=item MooseX::Storage;
+
+=item wtsi_clarity::util::types;
```
Comment (ke4) [inline src/perl/lib/wtsi_clarity/mq/messages/flowcell/control.pm]: Not used by this module

File: src/perl/lib/wtsi_clarity/mq/messages/flowcell/flowcell.pm [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+wtsi_clarity::mq::messages::flowcell::flowcell
+
+=head1 SYNOPSIS
+
+  my $flowcell = wtsi_clarity::mq::messages::flowcell::flowcell->new(
+    flowcell_barcode    => '123456778903',
+    flowcell_id         => '24-12345',
+    forward_read_length => 222,
+    reverse_read_length => 222,
+    lanes               => [{
```
Comment (ke4) [inline src/perl/lib/wtsi_clarity/mq/messages/flowcell/flowcell.pm]: missing the `updated_at` parameter

File: src/perl/lib/wtsi_clarity/mq/messages/flowcell/flowcell.pm [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        my @lanes = map { wtsi_clarity::mq::messages::flowcell::lane->new($_) } @{$_};
+        return \@lanes;
+      };
+
+my @defaults = ( is => 'ro', isa => 'Str', required => 1 );
+
+has ['flowcell_barcode',
+     'flowcell_id',
+     'forward_read_length',
+     'reverse_read_length',
```
Comment (ke4) [inline src/perl/lib/wtsi_clarity/mq/messages/flowcell/flowcell.pm]: `forward_read_length` and `reverse_read_length ` aren't they `Int` typed?

File: src/perl/lib/wtsi_clarity/mq/messages/flowcell/control.pm (hunk 1 above)
Comment (jbeast) [inline src/perl/lib/wtsi_clarity/mq/messages/flowcell/control.pm]: Didn't actually need it in the end so it's not added to the message. Will remove from synopsis.

File: src/perl/lib/wtsi_clarity/mq/messages/flowcell/sample.pm [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+=head1 SYNOPSIS
+
+  my $sample = wtsi_clarity::mq::messages::flowcell::sample->new(
+    tag_sequence               => 'ATAG',
+    tag_set_name               => 'Sanger_168tags - 10 mer tags',
+    pipeline_id_lims           => 'GCLP',
+    entity_type                => 'library_indexed',
+    bait_name                  => 'DDD_V5_plus',
+    sample_uuid                => '00000000-0000-0000-000000000',
+    study_uuid                 => '00000000-0000-0000-000000001',
```
Comment (ke4) [inline src/perl/lib/wtsi_clarity/mq/messages/flowcell/sample.pm]: No `study_uuid` definition in the module. If it is unneeded, then please remove it from the other synopsis, too.

File: src/perl/lib/wtsi_clarity/mq/messages/flowcell/sample.pm [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+=head1 DEPENDENCIES
+
+=over
+
+=item Moose
+
+=item MooseX::Storage;
+
+=item wtsi_clarity::util::types;
```
Comment (ke4) [inline src/perl/lib/wtsi_clarity/mq/messages/flowcell/sample.pm]: not used

File: src/perl/lib/wtsi_clarity/mq/messages/flowcell/sample.pm (hunk 5 above)
Comment (jbeast) [inline src/perl/lib/wtsi_clarity/mq/messages/flowcell/sample.pm]: It is not required but can be included so I think it should remain in the synopsis

**Pull request merged**